lib/view/sfViewCacheManager.class.php

```diff
@@ -1003,10 +1003,10 @@ public function getCurrentCacheKey()
   {
     $cacheKey = $this->routing->getCurrentInternalUri();
 
-    if ($getParameters = $this->request->getGetParameters())
+    if ($cacheKey && $getParameters = $this->request->getGetParameters())
```

Comment:
Is this meant to be here?
we will not consider the parameters if the $cacheKey is an empty string... do we have a test for this?

Reply:
Sorry there is no warning on concatenate on null ...

Reply:
The issue is on the `strpos($cacheKey, '?')` — I get the 3 times the error below for every page request for which the route doesn't exist (e.g. 404 case) that has query parameters in it.

```
PHP Deprecated:  strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in [...]/vendor/friendsofsymfony1/symfony1/lib/view/sfViewCacheManager.class.php on line 1008
```

There's no warning if there are no query parameters.

I suppose a more innocuous solution will be to cast the $cacheKey on `stripos()`, although I'm not sure it makes sense to be generating a cacheKey for a route that doesn't exist and only if it has query parameters. But perhaps in the context of this change, it's best to avoid any behavior changes.

Reply:
As the sfExecutionFilter can handle a null value

```
$uri = $this->context->getViewCacheManager()->getCurrentCacheKey();
if (null !== $uri && $this->context->getViewCacheManager()->hasActionCache($uri))
```

So _null_ is a valid value for getCurrentCacheKey() , 
Adding if($cacheKey...) check would break existing feature of enable caching if there are some GET parameters

So I revert this fix in order to keep the existing behavior.

We should need some test to check each possible $cacheKey value

Rebase Done

Reply:
We'll need to at least cast cacheKey on `strpos($cacheKey, '?')` or there are deprecation warnings on php8.1 with unnamed routes with parameters.

Reply:
Thanks @mentalstring , I applied your suggestion, by casting the $cacheKey to (string).
It should keep the previous behavior and fix the PHP8.1 warning.

Rebased !

@thePanz is it ok for you ? 
 
 
 Just for fun, I run some benchmark on casting to (string).
 On PHP8.1 I got a ~10% overhead to the strpos() call with the cast.